internal/driver/tempfile.go

```diff
@@ -25,9 +25,21 @@ import (
 func newTempFile(dir, prefix, suffix string) (*os.File, error) {
 	for index := 1; index < 10000; index++ {
 		path := filepath.Join(dir, fmt.Sprintf("%s%03d%s", prefix, index, suffix))
-		if _, err := os.Stat(path); err != nil {
-			return os.Create(path)
+		_, err := os.Stat(path)
+		if err == nil {
+			continue
 		}
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
```

Comment:
I dont think Stat() is needed. Just call OpenFile with `os.O_RDWR|os.O_CREATE|os.O_EXCL` directly and handle err as needed. If this is racy, checking Stat() prior to OpenFile does not guarantee success in all cases.

Reply:
Done. I thought about this and left the `os.Stat()` call to avoid spurious write accesses - somehow it felt better to only probe with a read rather than with a write. But it shouldn't matter as far as I can think of it, and less code is good.